### ISSUE TITLE
remove Pod::Parser from the core distribution

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#13194]
Migrated from [rt.perl.org#119439](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=119439) (status was 'pending release')

Searchable as RT119439$

**Comment by p5pRT** [comment]
### From @rjbs


We have been working \(slowwwly\) for years to remove Pod&#8203;::Parser from the core
perl distribution\.  At this point\, I believe the only blocker is Pod&#8203;::Checker\.
Pod&#8203;::Checker is waiting to be converted through the application of this patch&#8203;:

&nbsp;  https://rt.cpan.org/Ticket/Display.html?id=85442

Once that is done\, we will need to update the known problems file for the
podchecker test and we can add "use deprecate" to Pod&#8203;::Parser\.  \(Or so I
think\.\)

That should get done by perl 5\.20 so we can fully remove Pod&#8203;::Parser in perl
5\.22\.

\-\- 
rjbs

**Comment by p5pRT** [comment]
### From @rjbs

In the event that we are unable to get patches into Pod&#8203;::Checker to switch it to Pod&#8203;::Simple\, we 
will have to consider ejecting Pod&#8203;::Checker and renaming the Pod&#8203;::Checker\-with\-Pod&#8203;::Simple 
branch into Pod&#8203;::Simple&#8203;::Checker\, or something of the sort\, in order to move forward with this\.  I 
would rather not have it come to that\.

\-\- 
rjbs

**Comment by p5pRT** [comment]
@rjbs - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @cpansprout

On Mon Sep 02 19&#8203;:50&#8203;:21 2013\, rjbs wrote&#8203;:
> In the event that we are unable to get patches into Pod&#8203;::Checker to
> switch it to Pod&#8203;::Simple\, we
> will have to consider ejecting Pod&#8203;::Checker and renaming the
> Pod&#8203;::Checker\-with\-Pod&#8203;::Simple
> branch into Pod&#8203;::Simple&#8203;::Checker\, or something of the sort\, in order
> to move forward with this\.  I
> would rather not have it come to that\.

You say ‘we will have to’\, but is Pod&#8203;::Parser really that bad that we
have to take measures that would be confusing to users?

In other words\, why go to great lengths to remove it from core when that
time could be spent fixing core bugs? :\-\)

\-\- 

Father Chrysostomos

**Comment by p5pRT** [comment]
### From @rjbs

On Mon Sep 02 21&#8203;:36&#8203;:26 2013\, sprout wrote&#8203;:
> On Mon Sep 02 19&#8203;:50&#8203;:21 2013\, rjbs wrote&#8203;:
> > In the event that we are unable to get patches into Pod&#8203;::Checker to
> > switch it to Pod&#8203;::Simple\, we
> > will have to consider ejecting Pod&#8203;::Checker and renaming the
> > Pod&#8203;::Checker\-with\-Pod&#8203;::Simple
> > branch into Pod&#8203;::Simple&#8203;::Checker\, or something of the sort\, in order
> > to move forward with this\.  I
> > would rather not have it come to that\.
> 
> You say ‘we will have to’\, but is Pod&#8203;::Parser really that bad that we
> have to take measures that would be confusing to users?

You left out the word after "to\," which was "consider\."
 
We can worry about whether it's that bad or that hard to do once we're forced to consider\.  For 
now\, it is only a worry on the horizon\.

\-\- 
rjbs

**Comment by p5pRT** [comment]
### From @cpansprout

On Tue Sep 03 06&#8203;:10&#8203;:46 2013\, rjbs wrote&#8203;:
> On Mon Sep 02 21&#8203;:36&#8203;:26 2013\, sprout wrote&#8203;:
> > On Mon Sep 02 19&#8203;:50&#8203;:21 2013\, rjbs wrote&#8203;:
> > > In the event that we are unable to get patches into Pod&#8203;::Checker
> to
> > > switch it to Pod&#8203;::Simple\, we
> > > will have to consider ejecting Pod&#8203;::Checker and renaming the
> > > Pod&#8203;::Checker\-with\-Pod&#8203;::Simple
> > > branch into Pod&#8203;::Simple&#8203;::Checker\, or something of the sort\, in
> order
> > > to move forward with this\.  I
> > > would rather not have it come to that\.
> >
> > You say ‘we will have to’\, but is Pod&#8203;::Parser really that bad that
> we
> > have to take measures that would be confusing to users?
> 
> You left out the word after "to\," which was "consider\."
> 
> We can worry about whether it's that bad or that hard to do once we're
> forced to consider\.  For
> now\, it is only a worry on the horizon\.

Yes\, sorry\, I did misread it\.  But my response was only meant to be
tongue\-in\-cheek\. :\-\)

\-\- 

Father Chrysostomos

**Comment by p5pRT** [comment]
### From @xdg

On Thu\, Aug 22\, 2013 at 10&#8203;:25 PM\, Ricardo SIGNES
\<perlbug\-followup@&#8203;perl\.org> wrote&#8203;:
> We have been working \(slowwwly\) for years to remove Pod&#8203;::Parser from the core
> perl distribution\.  At this point\, I believe the only blocker is Pod&#8203;::Checker\.
> Pod&#8203;::Checker is waiting to be converted through the application of this patch&#8203;:
>
>   https://rt.cpan.org/Ticket/Display.html?id=85442

We're blocked on a patch not being applied since May?  :\-\(

I suggest setting a hard deadline for application of the patch and
release to CPAN with the alternative being that a volunteer co\-maint
will then be found and granted permissions to make it happen\.

\(I nominate mst to handle volunteer/PAUSE wrangling\, given his talents
at both\. :\-\)

David


\-\- 
David Golden \<xdg@&#8203;xdg\.me>
Take back your inbox\! → http&#8203;://www\.bunchmail\.com/
Twitter/IRC&#8203;: @&#8203;xdg

**Comment by p5pRT** [comment]
### From @jkeenan

On Tue Sep 03 09&#8203;:43&#8203;:17 2013\, xdg@&#8203;xdg\.me wrote&#8203;:
> On Thu\, Aug 22\, 2013 at 10&#8203;:25 PM\, Ricardo SIGNES
> \<perlbug\-followup@&#8203;perl\.org> wrote&#8203;:
> > We have been working \(slowwwly\) for years to remove Pod&#8203;::Parser from
> > the core
> > perl distribution\.  At this point\, I believe the only blocker is
> > Pod&#8203;::Checker\.
> > Pod&#8203;::Checker is waiting to be converted through the application of
> > this patch&#8203;:
> >
> > https://rt.cpan.org/Ticket/Display.html?id=85442
> 
> We're blocked on a patch not being applied since May?  :\-\(
> 

This indicates that CPAN \#85442 has been marked Resolved&#8203;:

https://rt.cpan.org/Public/Bug/Display.html?id=85442#txn-1265804

What are the next steps needed for this RT?

Thank you very much\.
Jim Keenan

**Comment by p5pRT** [comment]
### From @rjbs

On Tue Feb 18 18&#8203;:39&#8203;:52 2014\, jkeenan wrote&#8203;:
> 
> What are the next steps needed for this RT?
> 

The new Pod&#8203;::Checker emits different strings for the errors it finds\, based on Pod&#8203;::Simple rather than Pod&#8203;::Parser's warnings\.  Our pod exceptions \(the known errors file\) must be updated\.

Our podcheck test adds some tests of its own\, written toward Pod&#8203;::Parser rather than Pod&#8203;::Simple\.  Those tests need updating\.

Unless someone on a white horse rides in \(please oh please\)\, I will probably be doing this work at the QA Hackathon\.

\-\- 
rjbs

**Comment by p5pRT** [comment]
### From gottreu@gmail.com

On Feb 18\, 2014 8&#8203;:58 PM\, "Ricardo SIGNES via RT" \<perlbug\-followup@&#8203;perl\.org>
wrote&#8203;:
> The new Pod&#8203;::Checker emits different strings for the errors it finds\,
based on Pod&#8203;::Simple rather than Pod&#8203;::Parser's warnings\.  Our pod
exceptions \(the known errors file\) must be updated\.
>
> Our podcheck test adds some tests of its own\, written toward Pod&#8203;::Parser
rather than Pod&#8203;::Simple\.  Those tests need updating\.
>
> Unless someone on a white horse rides in \(please oh please\)\, I will
probably be doing this work at the QA Hackathon\.

As I haven't seen any horses\, I will take a swing at this\.

**Comment by p5pRT** [comment]
### From @rjbs

\* Brian Gottreu \<gottreu@&#8203;gmail\.com> \[2014\-02\-24T16&#8203;:33&#8203;:00\]
> As I haven't seen any horses\, I will take a swing at this\.

You will win my thanks\, and a round of tacos and drinks\.

\-\- 
rjbs

**Comment by p5pRT** [comment]
### From @khwilliamson

On 2/24/2014 6&#8203;:49 PM\, Ricardo Signes wrote&#8203;:
> \* Brian Gottreu \<gottreu@&#8203;gmail\.com> \[2014\-02\-24T16&#8203;:33&#8203;:00\]
>> As I haven't seen any horses\, I will take a swing at this\.
>
> You will win my thanks\, and a round of tacos and drinks\.
>

There is a version of podcheck that I thought rjbs was using that I 
thought was updated to use these; it certainly was updated to work with 
Marc Green's changes

**Comment by p5pRT** [comment]
### From gottreu@gmail.com

Version 1\.70 of Pod&#8203;::Checker removed the 'hyperlink' method which podcheck\.t uses\.  Where should that required functionality go?  Back in to Pod&#8203;::Checker without using Pod&#8203;::Hyperlink \(or whatever caused the removal\)\, into podcheck\.t\, or even up into Pod&#8203;::Simple?

**Comment by p5pRT** [comment]
### From @rjbs

\* Brian Gottreu via RT \<perlbug\-followup@&#8203;perl\.org> \[2014\-03\-05T01&#8203;:13&#8203;:29\]
> Version 1\.70 of Pod&#8203;::Checker removed the 'hyperlink' method which podcheck\.t
> uses\.  Where should that required functionality go?  Back in to Pod&#8203;::Checker
> without using Pod&#8203;::Hyperlink \(or whatever caused the removal\)\, into
> podcheck\.t\, or even up into Pod&#8203;::Simple?

I suggest putting it in podcheck\.t for now\.  If it turns out that it seems
generally useful\, it can be ported upward later\.

\-\- 
rjbs

**Comment by p5pRT** [comment]
### From @tonycoz

On Tue Mar 04 22&#8203;:13&#8203;:29 2014\, gottreu wrote&#8203;:
> Version 1\.70 of Pod&#8203;::Checker removed the 'hyperlink' method which
> podcheck\.t uses\.  Where should that required functionality go?  Back
> in to Pod&#8203;::Checker without using Pod&#8203;::Hyperlink \(or whatever caused
> the removal\)\, into podcheck\.t\, or even up into Pod&#8203;::Simple?

Hi Brian\,

Have you had time to work on this?

Otherwise I'll take a poke at it\.

Tony

**Comment by p5pRT** [comment]
### From @Smylers

Ricardo SIGNES writes&#8203;:

> We have been working \(slowwwly\) for years to remove Pod&#8203;::Parser from
> the core perl distribution\.  At this point\, I believe the only blocker
> is Pod&#8203;::Checker\.

Karl Williamson has heroically achieved this \(in RT \#116467\)\.

However\, perlpodspec is still promoting Pod&#8203;::Parser in a couple of
places\. Would readers be better served by being pointed at Pod&#8203;::Simple
instead?

This one can be addressed by simply changing the module name&#8203;:

&nbsp;  Authors of Pod formatters/processors should make every effort to avoid
&nbsp;  writing their own Pod parser\.  There are already several in CPAN\, with
&nbsp;  a wide range of interface styles \-\- and one of them\, Pod&#8203;::Parser\,
&nbsp;  comes with modern versions of Perl\.

But this one is trickier&#8203;:

&nbsp;  In parsing Pod\, a notably tricky part is the correct parsing of
&nbsp;  \(potentially nested\!\) formatting codes\.  Implementors should consult
&nbsp;  the code in the "parse\_text" routine in Pod&#8203;::Parser as an example of a
&nbsp;  correct implementation\.

What's the equivalent place in Pod&#8203;::Simple that would serve as an
example of how to get this right?

Smylers
\-\- 
http&#8203;://twitter\.com/Smylers2

**Comment by p5pRT** [comment]
### From @khwilliamson

On 05/31/2016 07&#8203;:30 AM\, Smylers wrote&#8203;:
> Ricardo SIGNES writes&#8203;:
>
>> We have been working \(slowwwly\) for years to remove Pod&#8203;::Parser from
>> the core perl distribution\.  At this point\, I believe the only blocker
>> is Pod&#8203;::Checker\.
>
> Karl Williamson has heroically achieved this \(in RT \#116467\)\.
>
> However\, perlpodspec is still promoting Pod&#8203;::Parser in a couple of
> places\. Would readers be better served by being pointed at Pod&#8203;::Simple
> instead?

Good catch
>
> This one can be addressed by simply changing the module name&#8203;:
>
>    Authors of Pod formatters/processors should make every effort to avoid
>    writing their own Pod parser\.  There are already several in CPAN\, with
>    a wide range of interface styles \-\- and one of them\, Pod&#8203;::Parser\,
>    comes with modern versions of Perl\.

Would you like to submit a patch for this?  If so\, I guarantee it will 
get applied\.

>
> But this one is trickier&#8203;:
>
>    In parsing Pod\, a notably tricky part is the correct parsing of
>    \(potentially nested\!\) formatting codes\.  Implementors should consult
>    the code in the "parse\_text" routine in Pod&#8203;::Parser as an example of a
>    correct implementation\.
>
> What's the equivalent place in Pod&#8203;::Simple that would serve as an
> example of how to get this right?

This one is tricky because the code in Pod&#8203;::Simple/BlackBox\.pm that 
mostly deals with this have various comments in it like&#8203;:

\# Are you really sure you want to read this code?
\# \- \- \-  Turn back now\!  Run away\!  \- \- \-
\#\#  stop reading now stop reading now stop reading now stop reading now stop
\#\#
\#\#                         HERE IT BECOMES REALLY SCARY
\#\#
\#\#  stop reading now stop reading now stop reading now stop reading now stop

So I don't know\, I'd be ok with leaving the Pod&#8203;::Parser reference in 
here  \(though I have never looked at its code actually\)
>
> Smylers
>

**Comment by p5pRT** [comment]
### From @khwilliamson

On Tue\, 31 May 2016 09&#8203;:56&#8203;:37 \-0700\, public@&#8203;khwilliamson\.com wrote&#8203;:
> On 05/31/2016 07&#8203;:30 AM\, Smylers wrote&#8203;:
> > Ricardo SIGNES writes&#8203;:
> >
> >> We have been working \(slowwwly\) for years to remove Pod&#8203;::Parser from
> >> the core perl distribution\.  At this point\, I believe the only blocker
> >> is Pod&#8203;::Checker\.
> >
> > Karl Williamson has heroically achieved this \(in RT \#116467\)\.
> >
> > However\, perlpodspec is still promoting Pod&#8203;::Parser in a couple of
> > places\. Would readers be better served by being pointed at Pod&#8203;::Simple
> > instead?
> 
> Good catch
> >
> > This one can be addressed by simply changing the module name&#8203;:
> >
> >    Authors of Pod formatters/processors should make every effort to avoid
> >    writing their own Pod parser\.  There are already several in CPAN\, with
> >    a wide range of interface styles \-\- and one of them\, Pod&#8203;::Parser\,
> >    comes with modern versions of Perl\.
> 
> Would you like to submit a patch for this?  If so\, I guarantee it will 
> get applied\.
> 
> >
> > But this one is trickier&#8203;:
> >
> >    In parsing Pod\, a notably tricky part is the correct parsing of
> >    \(potentially nested\!\) formatting codes\.  Implementors should consult
> >    the code in the "parse\_text" routine in Pod&#8203;::Parser as an example of a
> >    correct implementation\.
> >
> > What's the equivalent place in Pod&#8203;::Simple that would serve as an
> > example of how to get this right?
> 
> This one is tricky because the code in Pod&#8203;::Simple/BlackBox\.pm that 
> mostly deals with this have various comments in it like&#8203;:
> 
> \# Are you really sure you want to read this code?
> \# \- \- \-  Turn back now\!  Run away\!  \- \- \-
> \#\#  stop reading now stop reading now stop reading now stop reading now stop
> \#\#
> \#\#                         HERE IT BECOMES REALLY SCARY
> \#\#
> \#\#  stop reading now stop reading now stop reading now stop reading now stop
> 
> So I don't know\, I'd be ok with leaving the Pod&#8203;::Parser reference in 
> here  \(though I have never looked at its code actually\)
> >
> > Smylers
> >
> 

In lieu of a patch from Smylers\, I pushed one myself for the first reference to Pod&#8203;::Parser in perlpodspec as 67e5e059984f24c6c80a0f8278e40ce418d15c8b\.  I left the reference in that said to refer to it if you're writing a replacement\.  This is because of the comments in Pod&#8203;::Simple that are designed to scare people away\.

Pod&#8203;::Parser will not be removed in 5\.26\, as there is still a dependency on it\.  So\, I'm moving this ticket to block 5\.28 instead\.  The dependency is that it is the only way to extract just the pod from a larger file\.  I made significant headway in reimplementing that in Pod&#8203;::Simple\, but shelved it to work on higher priority things for 5\.26\.  I intend to finish it up once 5\.26 is in code freeze\.

\-\- 
Karl Williamson

**Comment by p5pRT** [comment]
### From @jkeenan

On Mon\, 09 Jan 2017 05&#8203;:47&#8203;:27 GMT\, khw wrote&#8203;:
> On Tue\, 31 May 2016 09&#8203;:56&#8203;:37 \-0700\, public@&#8203;khwilliamson\.com wrote&#8203;:
> > On 05/31/2016 07&#8203;:30 AM\, Smylers wrote&#8203;:
> > > Ricardo SIGNES writes&#8203;:
> > >
> > >> We have been working \(slowwwly\) for years to remove Pod&#8203;::Parser
> > >> from
> > >> the core perl distribution\.  At this point\, I believe the only
> > >> blocker
> > >> is Pod&#8203;::Checker\.
> > >
> > > Karl Williamson has heroically achieved this \(in RT \#116467\)\.
> > >
> > > However\, perlpodspec is still promoting Pod&#8203;::Parser in a couple of
> > > places\. Would readers be better served by being pointed at
> > > Pod&#8203;::Simple
> > > instead?
> >
> > Good catch
> > >
> > > This one can be addressed by simply changing the module name&#8203;:
> > >
> > > Authors of Pod formatters/processors should make every effort to
> > > avoid
> > > writing their own Pod parser\.  There are already several in CPAN\,
> > > with
> > > a wide range of interface styles \-\- and one of them\, Pod&#8203;::Parser\,
> > > comes with modern versions of Perl\.
> >
> > Would you like to submit a patch for this?  If so\, I guarantee it
> > will
> > get applied\.
> >
> > >
> > > But this one is trickier&#8203;:
> > >
> > > In parsing Pod\, a notably tricky part is the correct parsing of
> > > \(potentially nested\!\) formatting codes\.  Implementors should
> > > consult
> > > the code in the "parse\_text" routine in Pod&#8203;::Parser as an example
> > > of a
> > > correct implementation\.
> > >
> > > What's the equivalent place in Pod&#8203;::Simple that would serve as an
> > > example of how to get this right?
> >
> > This one is tricky because the code in Pod&#8203;::Simple/BlackBox\.pm that
> > mostly deals with this have various comments in it like&#8203;:
> >
> > \# Are you really sure you want to read this code?
> > \# \- \- \-  Turn back now\!  Run away\!  \- \- \-
> > \#\#  stop reading now stop reading now stop reading now stop reading
> > now stop
> > \#\#
> > \#\#                         HERE IT BECOMES REALLY SCARY
> > \#\#
> > \#\#  stop reading now stop reading now stop reading now stop reading
> > now stop
> >
> > So I don't know\, I'd be ok with leaving the Pod&#8203;::Parser reference in
> > here  \(though I have never looked at its code actually\)
> > >
> > > Smylers
> > >
> >
> 
> In lieu of a patch from Smylers\, I pushed one myself for the first
> reference to Pod&#8203;::Parser in perlpodspec as
> 67e5e059984f24c6c80a0f8278e40ce418d15c8b\.  I left the reference in
> that said to refer to it if you're writing a replacement\.  This is
> because of the comments in Pod&#8203;::Simple that are designed to scare
> people away\.
> 
> Pod&#8203;::Parser will not be removed in 5\.26\, as there is still a
> dependency on it\.  So\, I'm moving this ticket to block 5\.28 instead\.
> The dependency is that it is the only way to extract just the pod from
> a larger file\.  I made significant headway in reimplementing that in
> Pod&#8203;::Simple\, but shelved it to work on higher priority things for
> 5\.26\.  I intend to finish it up once 5\.26 is in code freeze\.

Karl\, do you have a battle plan for the removal of Pod\-Parser from core now that 5\.26\.0 is out the door?

Are there parts that could be delegated to other people?

Thank you very much\.
\-\- 
James E Keenan \(jkeenan@&#8203;cpan\.org\)

**Comment by p5pRT** [comment]
### From @khwilliamson

On 06/01/2017 07&#8203;:22 PM\, James E Keenan via RT wrote&#8203;:
> On Mon\, 09 Jan 2017 05&#8203;:47&#8203;:27 GMT\, khw wrote&#8203;:
>> On Tue\, 31 May 2016 09&#8203;:56&#8203;:37 \-0700\, public@&#8203;khwilliamson\.com wrote&#8203;:
>>> On 05/31/2016 07&#8203;:30 AM\, Smylers wrote&#8203;:
>>>> Ricardo SIGNES writes&#8203;:
>>>>
>>>>> We have been working \(slowwwly\) for years to remove Pod&#8203;::Parser
>>>>> from
>>>>> the core perl distribution\.  At this point\, I believe the only
>>>>> blocker
>>>>> is Pod&#8203;::Checker\.
>>>>
>>>> Karl Williamson has heroically achieved this \(in RT \#116467\)\.
>>>>
>>>> However\, perlpodspec is still promoting Pod&#8203;::Parser in a couple of
>>>> places\. Would readers be better served by being pointed at
>>>> Pod&#8203;::Simple
>>>> instead?
>>>
>>> Good catch
>>>>
>>>> This one can be addressed by simply changing the module name&#8203;:
>>>>
>>>> Authors of Pod formatters/processors should make every effort to
>>>> avoid
>>>> writing their own Pod parser\.  There are already several in CPAN\,
>>>> with
>>>> a wide range of interface styles \-\- and one of them\, Pod&#8203;::Parser\,
>>>> comes with modern versions of Perl\.
>>>
>>> Would you like to submit a patch for this?  If so\, I guarantee it
>>> will
>>> get applied\.
>>>
>>>>
>>>> But this one is trickier&#8203;:
>>>>
>>>> In parsing Pod\, a notably tricky part is the correct parsing of
>>>> \(potentially nested\!\) formatting codes\.  Implementors should
>>>> consult
>>>> the code in the "parse\_text" routine in Pod&#8203;::Parser as an example
>>>> of a
>>>> correct implementation\.
>>>>
>>>> What's the equivalent place in Pod&#8203;::Simple that would serve as an
>>>> example of how to get this right?
>>>
>>> This one is tricky because the code in Pod&#8203;::Simple/BlackBox\.pm that
>>> mostly deals with this have various comments in it like&#8203;:
>>>
>>> \# Are you really sure you want to read this code?
>>> \# \- \- \-  Turn back now\!  Run away\!  \- \- \-
>>> \#\#  stop reading now stop reading now stop reading now stop reading
>>> now stop
>>> \#\#
>>> \#\#                         HERE IT BECOMES REALLY SCARY
>>> \#\#
>>> \#\#  stop reading now stop reading now stop reading now stop reading
>>> now stop
>>>
>>> So I don't know\, I'd be ok with leaving the Pod&#8203;::Parser reference in
>>> here  \(though I have never looked at its code actually\)
>>>>
>>>> Smylers
>>>>
>>>
>>
>> In lieu of a patch from Smylers\, I pushed one myself for the first
>> reference to Pod&#8203;::Parser in perlpodspec as
>> 67e5e059984f24c6c80a0f8278e40ce418d15c8b\.  I left the reference in
>> that said to refer to it if you're writing a replacement\.  This is
>> because of the comments in Pod&#8203;::Simple that are designed to scare
>> people away\.
>>
>> Pod&#8203;::Parser will not be removed in 5\.26\, as there is still a
>> dependency on it\.  So\, I'm moving this ticket to block 5\.28 instead\.
>> The dependency is that it is the only way to extract just the pod from
>> a larger file\.  I made significant headway in reimplementing that in
>> Pod&#8203;::Simple\, but shelved it to work on higher priority things for
>> 5\.26\.  I intend to finish it up once 5\.26 is in code freeze\.
> 
> Karl\, do you have a battle plan for the removal of Pod\-Parser from core now that 5\.26\.0 is out the door?
> 
> Are there parts that could be delegated to other people?
> 
> Thank you very much\.
> 

I did not get to this during the code freeze\.  It involves deep work in 
Pod&#8203;::Simple\, and I don't think anybody else is doing anything on that 
module these days\.

It is \#3 on my todo list right now\, and I should get to it in a few weeks\.

**Comment by p5pRT** [comment]
### From @khwilliamson

On Thu\, 01 Jun 2017 20&#8203;:59&#8203;:47 \-0700\, public@&#8203;khwilliamson\.com wrote&#8203;:
> On 06/01/2017 07&#8203;:22 PM\, James E Keenan via RT wrote&#8203;:
> > On Mon\, 09 Jan 2017 05&#8203;:47&#8203;:27 GMT\, khw wrote&#8203;:
> >> On Tue\, 31 May 2016 09&#8203;:56&#8203;:37 \-0700\, public@&#8203;khwilliamson\.com wrote&#8203;:
> >>> On 05/31/2016 07&#8203;:30 AM\, Smylers wrote&#8203;:
> >>>> Ricardo SIGNES writes&#8203;:
> >>>>
> >>>>> We have been working \(slowwwly\) for years to remove Pod&#8203;::Parser
> >>>>> from
> >>>>> the core perl distribution\.  At this point\, I believe the only
> >>>>> blocker
> >>>>> is Pod&#8203;::Checker\.
> >>>>
> >>>> Karl Williamson has heroically achieved this \(in RT \#116467\)\.
> >>>>
> >>>> However\, perlpodspec is still promoting Pod&#8203;::Parser in a couple of
> >>>> places\. Would readers be better served by being pointed at
> >>>> Pod&#8203;::Simple
> >>>> instead?
> >>>
> >>> Good catch
> >>>>
> >>>> This one can be addressed by simply changing the module name&#8203;:
> >>>>
> >>>> Authors of Pod formatters/processors should make every effort to
> >>>> avoid
> >>>> writing their own Pod parser\.  There are already several in CPAN\,
> >>>> with
> >>>> a wide range of interface styles \-\- and one of them\, Pod&#8203;::Parser\,
> >>>> comes with modern versions of Perl\.
> >>>
> >>> Would you like to submit a patch for this?  If so\, I guarantee it
> >>> will
> >>> get applied\.
> >>>
> >>>>
> >>>> But this one is trickier&#8203;:
> >>>>
> >>>> In parsing Pod\, a notably tricky part is the correct parsing of
> >>>> \(potentially nested\!\) formatting codes\.  Implementors should
> >>>> consult
> >>>> the code in the "parse\_text" routine in Pod&#8203;::Parser as an example
> >>>> of a
> >>>> correct implementation\.
> >>>>
> >>>> What's the equivalent place in Pod&#8203;::Simple that would serve as an
> >>>> example of how to get this right?
> >>>
> >>> This one is tricky because the code in Pod&#8203;::Simple/BlackBox\.pm that
> >>> mostly deals with this have various comments in it like&#8203;:
> >>>
> >>> \# Are you really sure you want to read this code?
> >>> \# \- \- \-  Turn back now\!  Run away\!  \- \- \-
> >>> \#\#  stop reading now stop reading now stop reading now stop reading
> >>> now stop
> >>> \#\#
> >>> \#\#                         HERE IT BECOMES REALLY SCARY
> >>> \#\#
> >>> \#\#  stop reading now stop reading now stop reading now stop reading
> >>> now stop
> >>>
> >>> So I don't know\, I'd be ok with leaving the Pod&#8203;::Parser reference
> >>> in
> >>> here  \(though I have never looked at its code actually\)
> >>>>
> >>>> Smylers
> >>>>
> >>>
> >>
> >> In lieu of a patch from Smylers\, I pushed one myself for the first
> >> reference to Pod&#8203;::Parser in perlpodspec as
> >> 67e5e059984f24c6c80a0f8278e40ce418d15c8b\.  I left the reference in
> >> that said to refer to it if you're writing a replacement\.  This is
> >> because of the comments in Pod&#8203;::Simple that are designed to scare
> >> people away\.
> >>
> >> Pod&#8203;::Parser will not be removed in 5\.26\, as there is still a
> >> dependency on it\.  So\, I'm moving this ticket to block 5\.28 instead\.
> >> The dependency is that it is the only way to extract just the pod
> >> from
> >> a larger file\.  I made significant headway in reimplementing that in
> >> Pod&#8203;::Simple\, but shelved it to work on higher priority things for
> >> 5\.26\.  I intend to finish it up once 5\.26 is in code freeze\.
> >
> > Karl\, do you have a battle plan for the removal of Pod\-Parser from
> > core now that 5\.26\.0 is out the door?
> >
> > Are there parts that could be delegated to other people?
> >
> > Thank you very much\.
> >
> 
> I did not get to this during the code freeze\.  It involves deep work
> in
> Pod&#8203;::Simple\, and I don't think anybody else is doing anything on that
> module these days\.
> 
> It is \#3 on my todo list right now\, and I should get to it in a few
> weeks\.

I did not get to it until recently\.  I have a branch where podcheck\.t no longer relies on Pod&#8203;::Parser\.\.  However\, it turns out there are other dependencies on this module\, and I'm unsure what to do\.  Besides plain Pod&#8203;::Parser\, the distribution includes
Pod&#8203;::ParseUtils
Pod;&#8203;:InputObjects
Pod&#8203;::Select

and there are 3 tests that check for these&#8203;:
'lib/Pod/t/InputObjects\.t'
'lib/Pod/t/Select\.t'
'lib/Pod/t/utils\.t'

If I remove these tests\, everything passes\.  

Shall I remove these?

\-\- 
Karl Williamson

**Comment by p5pRT** [comment]
### From @cpansprout

On Sat\, 19 May 2018 14&#8203;:41&#8203;:40 \-0700\, khw wrote&#8203;:
> I did not get to it until recently\.  I have a branch where podcheck\.t
> no longer relies on Pod&#8203;::Parser\.\.  However\, it turns out there are
> other dependencies on this module\, and I'm unsure what to do\.  Besides
> plain Pod&#8203;::Parser\, the distribution includes
> Pod&#8203;::ParseUtils
> Pod;&#8203;:InputObjects
> Pod&#8203;::Select
> 
> and there are 3 tests that check for these&#8203;:
> 'lib/Pod/t/InputObjects\.t'
> 'lib/Pod/t/Select\.t'
> 'lib/Pod/t/utils\.t'
> 
> If I remove these tests\, everything passes\.
> 
> Shall I remove these?

Considering they begin with lib/Pod/\, they probably belong in the Pod\-Parser distribution \(and predate the reorganization of the modules in 5\.12 or so\)\.  Do they duplicate tests that Pod\-Parser has?  If so\, they can go\.  If not\, perhaps a patch to Pod\-Parser could be submitted\.  \(I admit I did not even look at them\.\)

I hope you are not intending to merge your branch until after 5\.28 is released\.

\-\- 

Father Chrysostomos

**Comment by p5pRT** [comment]
### From @khwilliamson

This has finally been fixed by c57372c9fa
\-\- 
Karl Williamson

**Comment by p5pRT** [comment]
@khwilliamson - Status changed from 'open' to 'pending release'

**Comment by kentfredric** [comment]
I'm just gonna point out, but this was removed without:
- a deprecation cycle (so there's no way CPAN would expose this missing in PREREQ_PM)
- a deprecation notice in Module::Metadata (so there's no way one could even ask if it was about to be removed from the tool who's job it is to advertise that )

```
# corelist CGI

Data for 2020-07-17
CGI was first released with perl 5.004, deprecated (will be CPAN-only) in v5.19.7 and removed from v5.21.0
# corelist Pod::Parser

Data for 2020-07-17
Pod::Parser was first released with perl v5.6.0 and removed from v5.31.1
```

The horse is bolted on that already, and we only noticed when we hit the first package that had this problem.

But there should be a system or reminder to prevent this happening in future.
